### PR TITLE
Alert refresh modal

### LINKED ISSUE #248: Show alert when leaving "Create Sarcophagus"
If a user attempts to leave the page, either by refreshing the page or by clicking on a header link, when in the process of the final step of "create sarcophagus" (the one with the steps on the side to create the TX):

1. They should be notified that they will lose all their progress. This  can be done via a modal alert box. See if we already have a modal setup (we may) that can be used for this. Otherwise, create a basic modal using Chakra.

If we don't have a modal and you need an example design, you can use the "hidden archs" modal in the figma as a template:

Image

2. The modal should have a "confirm" or "cancel" button. If the user confirms, then they should continue to whatever page they were going to and the state should be cleared. To clear the state:
a. Use the `clearSarcophagusState` method in the `useClearSarcophagusState` hook
b. Call `dispatch(goToStep(Step.NameSarcophagus))` to reset to the first step.

There should be a way to do this with React to detect when the component is unloading from the window. At that point the alert modal should show.

### PR DESCRIPTION
**Pull request by solidoracle**
### Summary

When in the process of the final step of "create sarcophagus" (the one with the steps on the side to create the TX), the user will be notified if it tries to refresh page (default alert) or when clicking on a header link (custom modal alert) that it will lose all its progress. 
State is cleared if user confirms refresh or new route.
Screenshot 2022-11-29 at 10 23 11
Screenshot 2022-11-29 at 11 23 17

As of today, React Router v6 doesn't ship with support for preventing transitions. I've implemented a workaround with a custom hook and modal.

### CODE REVIEW

File: src/features/embalm/stepContent/hooks/usePageBlock.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { History, Transition } from 'history';
+import { useContext, useEffect } from 'react';
+import { Navigator } from 'react-router';
+import { UNSAFE_NavigationContext as NavigationContext } from 'react-router-dom';
+
+export type ExtendNavigator = Navigator & Pick<History, 'block'>;
+
+export function useBlocker(blocker: (tx: Transition) => void) {
```
Comment (DarksightKellar) [inline src/features/embalm/stepContent/hooks/usePageBlock.tsx]: Doesn't seem like you need to export `ExtendNavigator` and `useBlocker`.

File: src/features/embalm/stepContent/components/PageBlockModal.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        <ModalContent
+          bgColor={colors.brand[0]}
+          border={'2px solid'}
+          borderColor={colors.brand[900]}
+          borderRadius={'0'}
+        >
+          <ModalHeader>Attention</ModalHeader>
+          <ModalCloseButton />
+
+          <ModalBody>Your progress will be lost if you change page.</ModalBody>
```
Comment (DarksightKellar) [inline src/features/embalm/stepContent/components/PageBlockModal.tsx]: Suggested edit: "Your progress will be lost if you move (or maybe, navigate) away from this page."

**Review by DarksightKellar — CHANGES_REQUESTED** [review]
Looks good. Will test soon. I just left a couple really minor suggested updates

File: src/features/embalm/stepContent/components/PageBlockModal.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        isCentered
+      >
+        <ModalOverlay />
+        <ModalContent
+          bgColor={colors.brand[0]}
+          border={'2px solid'}
+          borderColor={colors.brand[900]}
+          borderRadius={'0'}
+        >
+          <ModalHeader>Attention</ModalHeader>
```
Comment (DarksightKellar) [inline src/features/embalm/stepContent/components/PageBlockModal.tsx:84]: "Attention" -> "Warning", maybe?

Comment (solidoracle) [reply src/features/embalm/stepContent/components/PageBlockModal.tsx:84]: For consistency with the designs we have for another modal alert I kept 'Attention'
image

Comment (DarksightKellar) [reply src/features/embalm/stepContent/components/PageBlockModal.tsx:84]: Ah okay, fair enough.

**Review by DarksightKellar — APPROVED** [review]

**Review by DarksightKellar — APPROVED** [review]

**Review by sethhrbek — APPROVED** [review]

**Pull request merged**